nntrainer/cl_context.h

```diff
@@ -183,12 +191,33 @@ class ClContext {
     return entry->second(props);
   }
 
+  /**
+   * @brief register or return already present OpenCl kernel pointer
+   * @param kernel_string kernel implementation string
+   * @param kernel_name kernel name
+   * @return reference of std::shared_ptr<opencl::Kernel>
+   */
+  const SharedPtrClKernel &registerClKernel(std::string kernel_string,
+                                            std::string kernel_name);
+
+  /**
+   * @brief create OpenCl kernel
+   * @param kernel_string reference of implementation string
+   * @param kernel_name reference of kernel_name
+   * @param kernel_ptr_ reference of shared_ptr of Kernel
+   * @return true if successful, false otherwise
+   */
+  bool clCreateKernel(std::string &kernel_string, std::string &kernel_name,
```

Comment:
As I understood, it is not recommended call `clCreateKernel` directly, but should be call `registerClKernel`. Therefore, could we consider altering 'clCreateKernel' into a private member function?

Reply:
Updated the commit with this change